code/game/objects/structures/inflatable.dm

```diff
@@ -0,0 +1,316 @@
+		visible_message("<span class='danger'>[src] slowly deflates.</span>")
+		spawn(50)
+			var/obj/item/inflatable/R = new /obj/item/inflatable(loc)
+			src.transfer_fingerprints_to(R)
+			qdel(src)
+
+/obj/structure/inflatable/verb/hand_deflate()
+	set name = "Deflate"
+	set category = "Object"
+	set src in oview(1)
+
+	if(usr.stat || usr.restrained())
+		return
+
+	deflate()
+
+/obj/item/inflatable/door
+	name = "inflatable door"
+	desc = "A folded membrane which rapidly expands into a simple door on activation."
+	icon = 'icons/obj/inflatable.dmi'
+	icon_state = "folded_door"
+
+/obj/item/inflatable/door/attack_self(mob/user)
+	playsound(loc, 'sound/items/zip.ogg', 75, 1)
+	user << "<span class='notice'> You inflate [src].</span>"
+	var/obj/structure/inflatable/door/R = new /obj/structure/inflatable/door(user.loc)
+	src.transfer_fingerprints_to(R)
+	R.add_fingerprint(user)
+	qdel(src)
+
+/obj/structure/inflatable/door //Based on mineral door code
+	name = "inflatable door"
+	density = 1
+	anchored = 1
+	opacity = 0
+
+	icon = 'icons/obj/inflatable.dmi'
+	icon_state = "door_closed"
+
+	var/state = 0 //closed, 1 == open
+	var/isSwitchingStates = 0
+
+/obj/structure/inflatable/door/attack_ai(mob/user as mob) //those aren't machinery, they're just big fucking slabs of a mineral
+	if(isAI(user)) //so the AI can't open it
+		return
+	else if(isrobot(user)) //but cyborgs can
+		if(get_dist(user,src) <= 1) //not remotely though
+			return TryToSwitchState(user)
+
+/obj/structure/inflatable/door/attack_hand(mob/user as mob)
+	return TryToSwitchState(user)
+
+/obj/structure/inflatable/door/CanPass(atom/movable/mover, turf/target, height=0, air_group=0)
+	if(air_group)
+		return state
+	if(istype(mover, /obj/effect/beam))
+		return !opacity
+	return !density
+
+/obj/structure/inflatable/door/CanAtmosPass(turf/T)
+	return !density
+
+/obj/structure/inflatable/door/proc/TryToSwitchState(atom/user)
+	if(isSwitchingStates)
+		return
+	if(ismob(user))
+		var/mob/M = user
+		if(M.client)
+			if(iscarbon(M))
+				var/mob/living/carbon/C = M
+				if(!C.handcuffed)
+					SwitchState()
+			else
+				SwitchState()
+	else if(istype(user, /obj/mecha))
+		SwitchState()
+
+/obj/structure/inflatable/door/proc/SwitchState()
+	if(state)
+		Close()
+	else
+		Open()
+	air_update_turf(1)
+
+/obj/structure/inflatable/door/proc/Open()
+	isSwitchingStates = 1
+	//playsound(loc, 'sound/effects/stonedoor_openclose.ogg', 100, 1)
+	flick("door_opening",src)
+	sleep(10)
+	density = 0
+	opacity = 0
+	state = 1
+	update_icon()
+	isSwitchingStates = 0
+
+/obj/structure/inflatable/door/proc/Close()
+	isSwitchingStates = 1
+	//playsound(loc, 'sound/effects/stonedoor_openclose.ogg', 100, 1)
+	flick("door_closing",src)
+	sleep(10)
+	density = 1
+	opacity = 0
+	state = 0
+	update_icon()
+	isSwitchingStates = 0
+
+/obj/structure/inflatable/door/update_icon()
+	if(state)
+		icon_state = "door_open"
+	else
+		icon_state = "door_closed"
+
+/obj/structure/inflatable/door/deflate(var/violent=0)
```

Comment:
there is a way to avoid copypasting the entire proc just for the new path, just declare a var which has the path of the "broken" version of the wall, and make deflate proc use that no?